src/flamegpu/gpu/CUDASimulation.cu

```diff
@@ -1636,12 +1562,14 @@ void CUDASimulation::initialiseSingletons() {
         gpuErrchk(cudaStreamSynchronize(stream_0));
         if (DEVICE_HAS_RESET_CHECK == DEVICE_HAS_RESET_FLAG) {
             // Device has been reset, purge host mirrors of static objects/singletons
-            detail::curve::Curve::getInstance().purge(stream_0);
+            for (auto& ca : agent_map) {
+                ca.second->purgeCurve();
+            }
             if (singletons) {
                 singletons->rng.purge();
                 singletons->scatter.purge();
+                // @todo environment purge? Is this whole block still required?
```

Comment:
Does this need doign now?

Reply:
Good question.

Probably not, because if device gets reset after `CUDASimulation` has been init, there's probably going to be far more wrong than just the environment.

Reply:
Have removed the comment, but will leave this unresolved incase you feel different.

Reply:
Yeah I think the whole `CUDASimulation::initialiseSingletons()` could do with a small rework, all? of the CUDA device reset stuff is nolonger relevant.

Reply:
Have removed all trace of purge, but left in auto `cudaDeviceReset()`, this will require an extra review from you.